packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -149,15 +156,29 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
     );
 
     // Retrieve Unique ID
-    final savedFixedEndpointId =
-        await _endpointInfoStore.read(key: endpointIdStorageKey);
-    final fixedEndpointId = savedFixedEndpointId ?? const Uuid().v1();
-    if (savedFixedEndpointId == null) {
+    final endpointInformationVersion =
+        await _endpointInfoStore.read(key: _endpointInformationVersionKey);
+    String? retrievedEndpointId;
+
+    if (endpointInformationVersion == null) {
+      _endpointInfoStore.write(
+        key: _endpointInformationVersionKey,
+        value: _endpointInformationCurrentVersionValue,
+      );
+
```

Comment:
question: Should this block only execute on ios/android like:

```
if (!zIsWeb && (Platform.isAndroid || Platform.isIOS)) {
}
```

I mentioned in another comment I got an error from here on macos but I wonder if it would simplify things just to stop this logic far upstream (here) to simplify non ios/android platforms as much as possible.

Reply:
Thanks yes updated the logic